training/coqui_stt_training/evaluate_export.py

```diff
@@ -0,0 +1,179 @@
+import csv
+import os
+import sys
+import wave
+import io
+from functools import partial
+from multiprocessing import JoinableQueue, Manager, Process, cpu_count
+
+import numpy as np
+from coqui_stt_training.util.evaluate_tools import calculate_and_print_report
+from coqui_stt_training.util.config import (
+    Config,
+    initialize_globals_from_args,
+    initialize_globals_from_cli,
+    log_error,
+)
+from coqui_stt_training.util.audio import read_ogg_opus
+from six.moves import range, zip
+
+r"""
+This module should be self-contained:
+  - build libstt.so with TFLite:
+    - bazel build [...] --define=runtime=tflite [...] //native_client:libstt.so
+  - make -C native_client/python/ TFDIR=... bindings
+  - setup a virtualenv
+  - pip install native_client/python/dist/*.whl
+  - pip install -r requirements_eval_tflite.txt
+
+Then run with a TFLite model, a scorer and a CSV test file
+"""
+
+
+def tflite_worker(model, scorer, queue_in, queue_out, gpu_mask):
+    os.environ["CUDA_VISIBLE_DEVICES"] = str(gpu_mask)
+    try:
+        from stt import Model
+    except ModuleNotFoundError:
+        log_error('ImportError: No module named stt, use "pip install stt" ')
+        sys.exit(-1)
+
+    ds = Model(model)
+    if scorer:
+        ds.enableExternalScorer(scorer)
+
+    while True:
+        try:
+            msg = queue_in.get()
+
+            filename = msg["filename"]
+            filetype = filename.split("/")[-1].split(".")[-1]
+            if filetype == "wav":
+                fin = wave.open(filename, "rb")
+                audio = np.frombuffer(fin.readframes(fin.getnframes()), np.int16)
+                fin.close()
+            elif filetype == "opus":
+                with open(filename, "rb") as fin:
+                    audio_format, audio = read_ogg_opus(io.BytesIO(fin.read()))
+
+            decoded = ds.stt(audio)
+
+            queue_out.put(
+                {
+                    "wav": filename,
+                    "prediction": decoded,
+                    "ground_truth": msg["transcript"],
+                }
+            )
+        except FileNotFoundError as ex:
+            print("FileNotFoundError: ", ex)
+
+        print(queue_out.qsize(), end="\r")  # Update the current progress
+        queue_in.task_done()
+
+
+def main(args):
+    initialize_globals_from_args()
+    initialize_globals_from_cli()
+    manager = Manager()
+    work_todo = JoinableQueue()  # this is where we are going to store input data
+    work_done = manager.Queue()  # this where we are gonna push them out
+
+    processes = []
+    for i in range(args.proc):
+        worker_process = Process(
+            target=tflite_worker,
+            args=(args.model, args.scorer, work_todo, work_done, i),
+            daemon=True,
+            name="tflite_process_{}".format(i),
+        )
+        worker_process.start()  # Launch reader() as a separate python process
+        processes.append(worker_process)
+
+    print([x.name for x in processes])
```

Comment:
We should probably clean this up in a follow up issue. I created a wrapper for multiprocessing.Pool in `util/multiprocessing.py` which lets us handle this per-process initialization stuff in a cleaner way, and then the queuing stuff can be replaced with a simple `Pool.map` call. I filed https://github.com/coqui-ai/STT/issues/2050 to track, but it's not urgent.